diff --git a/node_modules/mp4box/dist/mp4box.all.js b/node_modules/mp4box/dist/mp4box.all.js
index f51ea7c..a3ef045 100644
--- a/node_modules/mp4box/dist/mp4box.all.js
+++ b/node_modules/mp4box/dist/mp4box.all.js
@@ -6907,7 +6907,7 @@ ISOFile.prototype.createSingleSampleMoof = function(sample) {
 	var trak = this.getTrackById(sample.track_id);
 	traf.add("tfhd").set("track_id", sample.track_id)
 					.set("flags", BoxParser.TFHD_FLAG_DEFAULT_BASE_IS_MOOF);
-	traf.add("tfdt").set("baseMediaDecodeTime", (sample.dts - trak.first_dts));
+	traf.add("tfdt").set("baseMediaDecodeTime", (sample.dts - (trak.first_dts || 0)));
 	traf.add("trun").set("flags", BoxParser.TRUN_FLAGS_DATA_OFFSET | BoxParser.TRUN_FLAGS_DURATION | 
 				 				  BoxParser.TRUN_FLAGS_SIZE | BoxParser.TRUN_FLAGS_FLAGS | 
 				 				  BoxParser.TRUN_FLAGS_CTS_OFFSET)
